File: src/DesignPattern.CSharp.Sigletons/Singleton.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DesignPattern.CSharp.Sigletons
{
    //延遲延遲實體化
    public class Singleton
    {
        private static Singleton singleton;
        private Singleton() { }
        public static Singleton GetInstance() => singleton ?? (singleton = new Singleton());
    }
}

File: src/DesignPattern.VbDotNet/Sigletons/Singleton.vb

Namespace DesignPattern.VbDotNet.Sigletons
    '延遲延遲實體化  VB.Net
    Public Class Singleton
        Private Shared singleton As Singleton
        Private Sub New()

        End Sub
        Public Shared Function GetInstance() As Singleton
            If singleton Is Nothing Then
                singleton = New Singleton()
            End If
            Return singleton
        End Function
    End Class

End Namespace
